pyvista/plotting/plotting.py

```diff
@@ -1925,9 +1943,9 @@ def prepare_mapper(scalars):
         self.add_actor(actor, reset_camera=reset_camera, name=name, culling=culling,
                        pickable=pickable, render=render)
 
-        # Add scalar bar if available
-        if stitle is not None and show_scalar_bar and (not rgb or _custom_opac):
-            self.add_scalar_bar(title=stitle, **scalar_bar_args)
+        # Only show scalar bar if there are scalars
+        if show_scalar_bar and scalars is not None:
+            self.add_scalar_bar(**scalar_bar_args)
```

Comment:
Is the logic equivalent here? Wondering about `_custom_opac`.

Reply:
Added in cad3364ac88f8a26481aa99deedaf4a37294395b.